@a2x/sdk/client dynamic x402 imports still require x402 under Next 16 Turbopack

## Problem

`@a2x/sdk/client` still requires consumers to install `x402` when bundled by Next.js 16 / Turbopack, even if the application only imports `A2XClient` and never enters an x402 payment code path.

This is similar to #134, which was fixed by #144 by changing the top-level static `x402/*` imports into dynamic imports. The current issue appears to be a residual bundler-resolution problem: Turbopack still resolves the dynamic import specifiers at build time.

## Reproduction

Observed while smoke-testing tarballed `@planetarium/oai2a2a-server` / `@planetarium/oai2a2a-codec` packages in a fresh temporary Next app.

Minimal shape:

```bash
npm init -y
npm install next@^16 react@^19 react-dom@^19 @a2x/sdk@0.13.1
```

Create an App Router route or module that imports `A2XClient`:

```ts
import { A2XClient } from "@a2x/sdk/client";

export const clientRef = A2XClient;
```

Run:

```bash
next build
```

## Actual result

Next 16.2.6 / Turbopack fails the build unless `x402` is installed directly in the consuming app:

```text
Module not found: Can't resolve 'x402/client'
Module not found: Can't resolve 'x402/shared'
Module not found: Can't resolve 'x402/types'
```

The import trace points through `@a2x/sdk/dist/...` and `@a2x/sdk/dist/client/index.js`.

Installing `x402@^1.2.0` in the consuming app makes the same build pass:

```bash
npm install x402@^1.2.0
next build
```

## Expected result

If `x402` remains an optional peer dependency, importing and bundling `@a2x/sdk/client` should not require `x402` unless the app actually uses x402 payment features.

Alternatively, if current bundlers cannot avoid resolving those specifiers, the package contract should make `x402` required for `@a2x/sdk/client` consumers, or expose a lightweight client entrypoint that does not reference `x402/*` at all.

## Environment

- `@a2x/sdk`: 0.13.1
- `next`: 16.2.6
- build output: `Next.js 16.2.6 (Turbopack)`
- Node: local macOS development environment

## Related

- #134: original static import / optional peer issue
- #144: changed static `x402/*` imports to dynamic imports and closed #134

The current reproduction happens after that fix, with dynamic imports still present.

--- a/.changeset/fuzzy-bundles-pay.md
+++ b/.changeset/fuzzy-bundles-pay.md
@@ -1,0 +1,5 @@
+---
+"@a2x/sdk": patch
+---
+
+Keep optional x402 runtime imports opaque so bundlers do not require x402 when apps only import `@a2x/sdk/client`.

--- a/packages/a2x/src/x402/client.ts
+++ b/packages/a2x/src/x402/client.ts
@@ -33,28 +33,44 @@ import type {
   X402SettleResponse,
 } from './types.js';
 
-// x402 is declared as an `optional` peer dep. Importing its runtime
-// helpers at the top level pulls them into the `@a2x/sdk/client` chunk
-// even on code paths that never touch x402, which breaks bundlers when
-// users haven't installed it (issue #134). Lazy-import inside the only
-// function that needs them so non-x402 consumers stay unaffected.
-let _x402Runtime:
-  | Promise<{
-      createPaymentHeader: typeof import('x402/client').createPaymentHeader;
-      safeBase64Decode: typeof import('x402/shared').safeBase64Decode;
-      PaymentPayloadSchema: typeof import('x402/types').PaymentPayloadSchema;
-    }>
-  | undefined;
+type X402Runtime = {
+  createPaymentHeader: (
+    signer: unknown,
+    x402Version: number,
+    requirement: unknown,
+  ) => Promise<string>;
+  safeBase64Decode: (input: string) => string;
+  PaymentPayloadSchema: { parse(input: unknown): unknown };
+};
+
+function x402Specifier(part: string): string {
+  return ['x402', part].join('/');
+}
+
+async function importOptionalPeer(
+  specifier: string,
+): Promise<Record<string, unknown>> {
+  // x402 is declared as an optional peer dependency. Keep the specifier
+  // computed so bundlers that statically inspect dynamic imports do not
+  // require x402 unless callers actually execute the signing path.
+  return import(/* @vite-ignore */ specifier);
+}
+
+let _x402Runtime: Promise<X402Runtime> | undefined;
 async function _loadX402Runtime() {
   if (!_x402Runtime) {
     _x402Runtime = (async () => {
       const [{ createPaymentHeader }, { safeBase64Decode }, { PaymentPayloadSchema }] =
         await Promise.all([
-          import('x402/client'),
-          import('x402/shared'),
-          import('x402/types'),
+          importOptionalPeer(x402Specifier('client')),
+          importOptionalPeer(x402Specifier('shared')),
+          importOptionalPeer(x402Specifier('types')),
         ]);
-      return { createPaymentHeader, safeBase64Decode, PaymentPayloadSchema };
+      return {
+        createPaymentHeader: createPaymentHeader as X402Runtime['createPaymentHeader'],
+        safeBase64Decode: safeBase64Decode as X402Runtime['safeBase64Decode'],
+        PaymentPayloadSchema: PaymentPayloadSchema as X402Runtime['PaymentPayloadSchema'],
+      };
     })();
   }
   return _x402Runtime;
@@ -157,9 +173,9 @@ export async function signX402Payment(
     await _loadX402Runtime();
 
   const header = await createPaymentHeader(
-    options.signer as unknown as Parameters<typeof createPaymentHeader>[0],
+    options.signer,
     required.x402Version,
-    requirement as unknown as Parameters<typeof createPaymentHeader>[2],
+    requirement,
   );
 
   // `createPaymentHeader` returns a base64-encoded PaymentPayload for HTTP

